source/History.js

```diff
@@ -0,0 +1,389 @@
+(function (enyo, scope) {
+	/**
+	* {@link moon.History} is a decorator that eables back key feature..
+	* When the object get back key input like pressing back key in R/C
+	* or history.back() with window.history object, it calls back key handler.
+	*
+	*
+	*/
+	moon.HistorySupport = {
+		/**
+		* @private
+		*/
+		name: 'HistorySupport',
+
+		published: {
+			/**
+			* When true, pressing back key makes panels close opened drawer
+			*
+			* @type {Bollean}
+			* @default true
+			* @public
+			*/
+			allowBackKey: true
+		},
+
+		/**
+		* @private
+		*/
+		events: {
+			onPushBackHistory: ''
+		},
+
+		/**
+		* @private
+		*/
+		handlers: {
+			/**
+			* Hanlder for back key input.
+			* To use custom behavior for back key, you can modify this handler.
+			*/
+			onPushBackHistory: 'pushBackHistory'
+		},
+
+		/**
+		* @private
+		*/
+		pushBackHistory: function () {
```

Comment:
If we want to allow developers to override this in a given control (see above comment), I think we need to modify this to something like the following:
```
pushBackHistory: enyo.inherit(function (sup) {
			return function () {
				if (!sup.apply(this, arguments)) {
					moon.History.pushBackHistory(this, this.backKeyHandler || moon.History.backKeyHandler);
					return true;
				}
			};
		})
```